### PR TITLE
enhance FOHeader with language change

### PR DESCRIPTION
**Pull request by pranotoism**
### Detailed description of changes
- Enhance FOHeader with language change
- Enhance FOHeader interface 
- Add new type Header with language change in FLDefault and FLAuth stories
- Add Flags for english and bahasa
...

### Recommendations on how to test the changes
- Functionality check
...

### Checklist for developer

- [x] The code builds clean without any errors or warnings
- [x] The code does not contain commented out code
- [x] The code does not log anything to console
- [x] I have thoroughly tested the new code and any adjacent features it may affect

### Link to related issue(s)
https://app.zenhub.com/workspaces/marketplace---cygnus-62e793515a49bf0011835ff6/issues/hzn-one/marketplace-delivery/76
...

#### Before
Screenshot from 2022-08-22 10-56-30

...

#### After
Screenshot from 2022-08-22 10-56-43
Screenshot from 2022-08-22 19-04-10

...

### CODE REVIEW

**Comment by bryanprimus** [comment]
ini pake input base ya. kalau select nya dibuka bentuk nya kaya gmana bay ? bisa tolong di ss kah ?

**Comment by pranotoism** [comment]
> ini pake input base ya. kalau select nya dibuka bentuk nya kaya gmana bay ? bisa tolong di ss kah ?

@bryantobing12 no problem. gambar ada di deskripsi ya bry

**Comment by bryanprimus** [comment]
sepertinya bisa di achieve tanpa harus pake `@mui/base`. Bisa tolong coba kaya ini bay. https://github.com/HZN-one/marketplace/blob/31ee2e591eac049eea4f61ced4de8dab67d3e4bc/src/Components/Layouts/BasicLayout.tsx#L78-L97

Itu tinggal ngubah adornment nya jadi flag.

**Comment by pranotoism** [comment]
> sepertinya bisa di achieve tanpa harus pake `@mui/base`. Bisa tolong coba kaya ini bay. https://github.com/HZN-one/marketplace/blob/31ee2e591eac049eea4f61ced4de8dab67d3e4bc/src/Components/Layouts/BasicLayout.tsx#L78-L97
> 
> Itu tinggal ngubah adornment nya jadi flag.

@bryantobing12 no worries. updated ya bry

File: src/Components/Organisms/FOHeader/FOHeader.interface.ts [hunk 1]
```diff
@@ -8,6 +8,10 @@ interface IFOHeaderGeneral {
   logo?: boolean;
   companyLogo?: ReactChild;
   headerType: "auth" | "default";
+  language?: {
+    value: "EN" | "ID";
+    onChange?: (event: any) => void;
```
Comment (bryanprimus) [inline src/Components/Organisms/FOHeader/FOHeader.interface.ts]: kindly use the proper type instead of `any`.

**Review by bryanprimus — APPROVED** [review]

**Pull request merged**